controls/views.py

```diff
@@ -138,6 +138,47 @@ def control(request, catalog_key, cl_id):
     }
     return render(request, "controls/detail.html", context)
 
+
+def raise_404_if_system_permissions_not(request, system, permission='view_system', not_callback=None):
+    """Raise an HTTP 404 exception after calling callback(permission, action, user_message)  if the user doesn't have the system permission"""
+    if not request.user.has_perm(permission, system):
+        if not_callback:
+            not_callback(permission, 'permission_denied',
+                         'You do not have permission to change this system')
+
+        raise Http404(f'Access denied:  You do not have permission')
+        # 404 instead of 403 + vague message  to not leak info.
+        #   Message only conveyed when DEBUG True
+
+
+def log_system_action(request, system, action, message={}):
```

Comment:
Note:  Used in a parallel PR in progress ...

Just didn't spend time on combining only to revert in that later PR ; )